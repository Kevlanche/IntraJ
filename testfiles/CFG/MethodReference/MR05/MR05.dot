digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"; "MethodDecl[23]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[17]"; "List[18]"; "List[19]"; "Opt[20]"; "Exit[21]"; "Entry[24]"; "Modifiers[25]"; "PrimitiveTypeAccess[29]"; "List[30]"; "List[36]"; "Opt[37]"; "Exit[38]"}
{ rank = same; "List[15]"; "Block[21]"; "List[26]"; "ParameterDeclaration[31]"; "Block[38]"}
{ rank = same; "Modifier[16]"; "List[22]"; "Modifier[27]"; "Modifier[28]"; "Modifiers[32]"; "ArrayTypeAccess[34]"; "List[39]"}
{ rank = same; "List[33]"; "TypeAccess[35]"; "VarDeclStmt[40]"; "ExprStmt[62]"}
{ rank = same; "Modifiers[41]"; "TypeAccess[43]"; "List[44]"; "Dot[63]"}
{ rank = same; "List[42]"; "VariableDeclarator[45]"; "VarAccess[64]"; "MethodAccess[65]"}
{ rank = same; "List[46]"; "Opt[47]"; "List[67]"; "List[68]"; "List[69]"; "UncheckedExceptions[66]"}
{ rank = same; "ClassInstanceExpr[48]"}
{ rank = same; "TypeAccess[49]"; "List[50]"; "Opt[59]"; "List[60]"; "List[61]"}
{ rank = same; "ExprMethodReference[51]"}
{ rank = same; "List[52]"; "ClassInstanceExpr[53]"}
{ rank = same; "TypeAccess[54]"; "List[55]"; "Opt[56]"; "List[57]"; "List[58]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < MR2 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < bar >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[16]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[17]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[20]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[21]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[22]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[21]" [label="Exit\n ", fillcolor=white   style=filled  ]
"MethodDecl[23]" [label="MethodDecl\n < main >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[24]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[25]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[27]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[28]" [label="Modifier\n < static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[29]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[31]" [label="ParameterDeclaration\n < args >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[32]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ArrayTypeAccess[34]" [label="ArrayTypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[35]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[37]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[38]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[40]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[41]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[43]" [label="TypeAccess\n < Thread >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[45]" [label="VariableDeclarator\n < Thread t2 >", fillcolor=white   style=filled  ]
"List[46]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[47]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassInstanceExpr[48]" [label="ClassInstanceExpr\n <Thread>", fillcolor=white   style=filled  ]
"TypeAccess[49]" [label="TypeAccess\n < Thread >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprMethodReference[51]" [label="ExprMethodReference\n < bar >", fillcolor=white   style=filled  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassInstanceExpr[53]" [label="ClassInstanceExpr\n <MR2>", fillcolor=white   style=filled  ]
"TypeAccess[54]" [label="TypeAccess\n < MR2 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[55]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[56]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[57]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[58]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[59]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[60]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[62]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[63]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[64]" [label="VarAccess\n < t2 >", fillcolor=white   style=filled  ]
"MethodAccess[65]" [label="MethodAccess\n < start >", fillcolor=white   style=filled  ]
"List[67]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[68]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[69]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[66]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[38]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"List[15]" -> "Modifier[16]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[18]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[19]" [style=dashed, color=gray]
"Block[21]" -> "List[22]" [style=dashed, color=gray]
"Opt[20]" -> "Block[21]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[20]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[21]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"MethodDecl[23]" -> "Entry[24]" [style=dashed, color=gray]
"List[26]" -> "Modifier[27]" [style=dashed, color=gray]
"List[26]" -> "Modifier[28]" [style=dashed, color=gray]
"Modifiers[25]" -> "List[26]" [style=dashed, color=gray]
"MethodDecl[23]" -> "Modifiers[25]" [style=dashed, color=gray]
"MethodDecl[23]" -> "PrimitiveTypeAccess[29]" [style=dashed, color=gray]
"Modifiers[32]" -> "List[33]" [style=dashed, color=gray]
"ParameterDeclaration[31]" -> "Modifiers[32]" [style=dashed, color=gray]
"ArrayTypeAccess[34]" -> "TypeAccess[35]" [style=dashed, color=gray]
"ParameterDeclaration[31]" -> "ArrayTypeAccess[34]" [style=dashed, color=gray]
"List[30]" -> "ParameterDeclaration[31]" [style=dashed, color=gray]
"MethodDecl[23]" -> "List[30]" [style=dashed, color=gray]
"MethodDecl[23]" -> "List[36]" [style=dashed, color=gray]
"Modifiers[41]" -> "List[42]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "Modifiers[41]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "TypeAccess[43]" [style=dashed, color=gray]
"VariableDeclarator[45]" -> "List[46]" [style=dashed, color=gray]
"ClassInstanceExpr[48]" -> "TypeAccess[49]" [style=dashed, color=gray]
"ExprMethodReference[51]" -> "List[52]" [style=dashed, color=gray]
"ClassInstanceExpr[53]" -> "TypeAccess[54]" [style=dashed, color=gray]
"ClassInstanceExpr[53]" -> "List[55]" [style=dashed, color=gray]
"ClassInstanceExpr[53]" -> "Opt[56]" [style=dashed, color=gray]
"ClassInstanceExpr[53]" -> "List[57]" [style=dashed, color=gray]
"ClassInstanceExpr[53]" -> "List[58]" [style=dashed, color=gray]
"ExprMethodReference[51]" -> "ClassInstanceExpr[53]" [style=dashed, color=gray]
"List[50]" -> "ExprMethodReference[51]" [style=dashed, color=gray]
"ClassInstanceExpr[48]" -> "List[50]" [style=dashed, color=gray]
"ClassInstanceExpr[48]" -> "Opt[59]" [style=dashed, color=gray]
"ClassInstanceExpr[48]" -> "List[60]" [style=dashed, color=gray]
"ClassInstanceExpr[48]" -> "List[61]" [style=dashed, color=gray]
"Opt[47]" -> "ClassInstanceExpr[48]" [style=dashed, color=gray]
"VariableDeclarator[45]" -> "Opt[47]" [style=dashed, color=gray]
"List[44]" -> "VariableDeclarator[45]" [style=dashed, color=gray]
"VarDeclStmt[40]" -> "List[44]" [style=dashed, color=gray]
"List[39]" -> "VarDeclStmt[40]" [style=dashed, color=gray]
"Dot[63]" -> "VarAccess[64]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[67]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[68]" [style=dashed, color=gray]
"MethodAccess[65]" -> "List[69]" [style=dashed, color=gray]
"MethodAccess[65]" -> "UncheckedExceptions[66]" [style=dashed, color=gray]
"Dot[63]" -> "MethodAccess[65]" [style=dashed, color=gray]
"ExprStmt[62]" -> "Dot[63]" [style=dashed, color=gray]
"List[39]" -> "ExprStmt[62]" [style=dashed, color=gray]
"Block[38]" -> "List[39]" [style=dashed, color=gray]
"Opt[37]" -> "Block[38]" [style=dashed, color=gray]
"MethodDecl[23]" -> "Opt[37]" [style=dashed, color=gray]
"MethodDecl[23]" -> "Exit[38]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[21]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "Exit[21]" [color=blue, constraint=false, xlabel=" "] 
"Exit[38]" -> "MethodAccess[65]" [color=red, constraint=false]
"MethodAccess[65]" -> "Exit[38]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[65]" -> "VarAccess[64]" [color=red, constraint=false]
"VarAccess[64]" -> "MethodAccess[65]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[64]" -> "VariableDeclarator[45]" [color=red, constraint=false]
"VariableDeclarator[45]" -> "VarAccess[64]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[45]" -> "ClassInstanceExpr[48]" [color=red, constraint=false]
"ClassInstanceExpr[48]" -> "VariableDeclarator[45]" [color=blue, constraint=false, xlabel=" "] 
"ClassInstanceExpr[48]" -> "ExprMethodReference[51]" [color=red, constraint=false]
"ExprMethodReference[51]" -> "ClassInstanceExpr[48]" [color=blue, constraint=false, xlabel=" "] 
"ExprMethodReference[51]" -> "ClassInstanceExpr[53]" [color=red, constraint=false]
"ClassInstanceExpr[53]" -> "ExprMethodReference[51]" [color=blue, constraint=false, xlabel=" "] 
"ClassInstanceExpr[53]" -> "Entry[24]" [color=red, constraint=false]
"Entry[24]" -> "ClassInstanceExpr[53]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"MethodDecl[12]" -> "MethodDecl[23]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[17]" -> "List[18]" -> "List[19]" -> "Opt[20]" -> "Exit[21]" [style=invis]
"Entry[24]" -> "Modifiers[25]" -> "PrimitiveTypeAccess[29]" -> "List[30]" -> "List[36]" -> "Opt[37]" -> "Exit[38]" [style=invis]
"Modifier[27]" -> "Modifier[28]" [style=invis]
"Modifiers[32]" -> "ArrayTypeAccess[34]" [style=invis]
"VarDeclStmt[40]" -> "ExprStmt[62]" [style=invis]
"Modifiers[41]" -> "TypeAccess[43]" -> "List[44]" [style=invis]
"List[46]" -> "Opt[47]" [style=invis]
"TypeAccess[49]" -> "List[50]" -> "Opt[59]" [style=invis]
"List[52]" -> "ClassInstanceExpr[53]" [style=invis]
"TypeAccess[54]" -> "List[55]" -> "Opt[56]" [style=invis]
"VarAccess[64]" -> "MethodAccess[65]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
72", fillcolor=1]
Edges [label="Total Nr. Edges
87", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
8", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}