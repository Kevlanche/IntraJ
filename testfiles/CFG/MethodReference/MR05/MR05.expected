NodeName: Entry[13] Successors: [Exit[21]]
NodeName: Exit[21] Successors: []
NodeName: Entry[24] Successors: [ClassInstanceExpr[53]]
NodeName: ClassInstanceExpr[53] Successors: [ExprMethodReference[51]]
NodeName: ExprMethodReference[51] Successors: [ClassInstanceExpr[48]]
NodeName: ClassInstanceExpr[48] Successors: [VariableDeclarator[45]]
NodeName: VariableDeclarator[45] Successors: [VarAccess[64]]
NodeName: VarAccess[64] Successors: [MethodAccess[65]]
NodeName: MethodAccess[65] Successors: [Exit[38]]
NodeName: Exit[38] Successors: []

NodeName: Exit[21] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: Exit[38] Predecessor: [MethodAccess[65]]
NodeName: MethodAccess[65] Predecessor: [VarAccess[64]]
NodeName: VarAccess[64] Predecessor: [VariableDeclarator[45]]
NodeName: VariableDeclarator[45] Predecessor: [ClassInstanceExpr[48]]
NodeName: ClassInstanceExpr[48] Predecessor: [ExprMethodReference[51]]
NodeName: ExprMethodReference[51] Predecessor: [ClassInstanceExpr[53]]
NodeName: ClassInstanceExpr[53] Predecessor: [Entry[24]]
NodeName: Entry[24] Predecessor: []
